### DISCUSSION TITLE
File Manager types for a function table

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1248] [General]
An Interlisp program I'm writing has a global variable bound to a function table, an association list with pairs whose key is a symbol and whose value is a function. The initialization of the table will look like this:

```
(SETQ FUNTAB `((KEY1 ,(FUNCTION FUN1)) ... (KEYn ,(FUNCTION FUNn))))
```

where the functions `FUN1 ... FUNn` are defined in the same symbolic file.

What's the most appropriate File Manater type for managing the table and storing it in the program's symbolic file? Is `VARS` enough? Do I need `UGLYVARS`?

**Comment by nbriggs** [comment]
`ALISTS` is a file package command/type that will let you define which keys should be saved from a given ALIST.  If you want to save the whole thing, VARS should be fine.  The value of FUNTAB will look like `((KEY1 FUN1) ... (KEYn FUNn))` so there's nothing special that would suggest UGLYVARS (that's for when PRINT and READ are not inverses of each other).

**Comment by nbriggs** [comment]
Also see the IRM entry for `FUNCTION`...

**Comment by pamoroso** [comment]
Thanks, I'll go with `VARS`. As for initializing the table, I wasn't sure how to properly reference functions.
